x-pack/plugins/maps/public/selectors/map_selectors.js

```diff
@@ -163,3 +165,11 @@ export const getUniqueIndexPatternIds = createSelector(
 );
 
 export const getTemporaryLayers = createSelector(getLayerList, (layerList) => layerList.filter(layer => layer.isTemporary()));
+
+export const hasDirtyState = createSelector(getLayerListRaw, (layerListRaw) => {
```

Comment:
Does this method need to look at all layers? Couldn't just check that the selected layer is dirty?